File: frontend/src/types/part_parameter.ts
export interface PartParameter {
  id: number;
  component_id: number;
  name: string;
  symbol: string | null;
  value_text: string | null;
  value_min: number | null;
  value_typical: number | null;
  value_max: number | null;
  unit: string | null;
  group_name: string | null;
  sort_order: number;
  created_at: string;
}

export interface PartParameterCreate {
  name: string;
  symbol?: string;
  value_text?: string;
  value_min?: number;
  value_typical?: number;
  value_max?: number;
  unit?: string;
  group_name?: string;
  sort_order?: number;
}

export interface PartParameterUpdate extends Partial<PartParameterCreate> {}

File: backend/app/schemas/part_parameter.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class PartParameterBase(BaseModel):
    name: str
    symbol: Optional[str] = None
    value_text: Optional[str] = None
    value_min: Optional[float] = None
    value_typical: Optional[float] = None
    value_max: Optional[float] = None
    unit: Optional[str] = None
    group_name: Optional[str] = None
    sort_order: int = 0


class PartParameterCreate(PartParameterBase):
    pass


class PartParameterUpdate(BaseModel):
    name: Optional[str] = None
    symbol: Optional[str] = None
    value_text: Optional[str] = None
    value_min: Optional[float] = None
    value_typical: Optional[float] = None
    value_max: Optional[float] = None
    unit: Optional[str] = None
    group_name: Optional[str] = None
    sort_order: Optional[int] = None


class PartParameterResponse(PartParameterBase):
    id: int
    component_id: int
    created_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)
